app/interactions/actions/a_and_aaaa_email_validation.rb

```diff
@@ -16,19 +16,28 @@ def check_for_records_value(email:, value:)
     end
 
     def resolve_a_and_aaaa_records(dns_servers:, email_domain:, value:)
-      Resolv::DNS.open({ nameserver: dns_servers }) do |dns|
-        dns.timeouts = ENV['a_and_aaaa_validation_timeout'].to_i || 1
-        ress = nil
+      Resolv::DNS.open(nameserver: dns_servers, ndots: 1, search: []) do |dns|
+        dns.timeouts = (ENV['a_and_aaaa_validation_timeout'] || 1).to_i
 
         case value
         when 'A'
-          ress = dns.getresources email_domain, Resolv::DNS::Resource::IN::A
+          resolve_a_records(dns: dns, domain: email_domain)
         when 'AAAA'
-          ress = dns.getresources email_domain, Resolv::DNS::Resource::IN::AAAA
+          resolve_aaaa_records(dns: dns, domain: email_domain)
+        else
+          []
         end
-
-        ress.map(&:address)
       end
     end
+
+    def resolve_a_records(dns:, domain:)
```

Comment:
domain is not clear, rename to hostname, because Resolv::DNS::Resource::IN::A and Resolv::DNS::Resource::IN::AAAA in context of validation check hostname of email, but not domain itself